tests/metrics_test_utils.go

```diff
@@ -0,0 +1,59 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var regexpForMetrics = map[string]string{
+	"total_rejected_vms": `total_rejected_vms ([0-9]+)`,
+}
+
+func intMetricValue(metricName string, metricsPort uint16, pod *v1.Pod) int {
+	cfg, err := config.GetConfig()
+	Expect(err).ToNot(HaveOccurred())
+	coreClient, err := kubernetes.NewForConfig(cfg)
+	Expect(err).ToNot(HaveOccurred())
+
+	portForwarder := NewPortForwarder(cfg, coreClient.CoreV1().RESTClient())
+	conn, _ := portForwarder.Connect(pod, metricsPort)
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return conn, nil
+			},
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+	resp, err := client.Get(fmt.Sprintf("https://localhost:%d/metrics", metricsPort))
+	Expect(err).To(BeNil(), "Can't get metrics from %s", pod.Name)
```

Comment:
Instead of `.To(BeNil())`, you can use `.To(Succeed())` or `.ToNot(HaveOccured())`. These matchers expect an error type, and their message in case of failure is more clear.

I didn't know the second argument to `.To()` could be a format string, nice.

Reply:
done